### DISCUSSION TITLE
useQuery hook/callback?

### DISCUSSION THREAD

**Discussion by eensander** [wobsoriano/trpc-nuxt#121] [General]
Hello,

So I've been experimenting with this package and the integration with Nuxt seems flawless. However, I'm starting to bump my head regarding the useQuery function: how in the world do I know when the `.refresh()` is finished, or when data changes? I've been searching but can't find anything related to my problem.
I tried using Vue's/Nuxt's `watch(query, () => {})`, but get an error: `Invalid watch source: Promise { <state>: "pending" }`.

To me seemingly trivial args such as `useQuery(..., { onSuccess() { ... } })`, don't seem to exist. Note that I'm fairly new to (t)RPC, though I'd say such a callback to a refreshable querable object is pretty evident, not?

Please let me know what patterns we use to solve these use cases. Thanks a bunch!

**Comment by eensander** [comment]
Solved by watching actual properties of `query`, like `.data`, `.pending`. etc.

**Reply by wobsoriano** [reply]
Yes, that's correct.
